src/ElasticSearchRepository.php

```diff
@@ -44,34 +47,42 @@ public function __construct(
         Client $client,
         Serializer $serializer,
         string $index,
-        string $class,
+        string $class_type,
         array $notAnalyzedFields = []
     ) {
         $this->client = $client;
         $this->serializer = $serializer;
         $this->index = $index;
-        $this->class = $class;
+        $this->class_type = $class_type;
         $this->notAnalyzedFields = $notAnalyzedFields;
     }
 
     /**
-     * {@inheritdoc}
+     * @param Identifiable $data
+     * @throws AssertionFailedException
      */
     public function save(Identifiable $data): void
     {
-        Assertion::isInstanceOf($data, $this->class);
+        if (!class_exists($this->class_type)) {
+            throw new InvalidArgumentException(
+                "The class type provided ({$this->class_type}) does not exists."
+            );
+        }
+
+        Assertion::isInstanceOf(
```

Comment:
The more I think of this the more I favor a custom exception for this. Maybe later.